File: Phonebook.Frontend/src/app/shared/models/classes/Person.ts
import { Room } from 'src/app/shared/models/classes/Room';
import { PersonStatus } from 'src/app/shared/models/enumerables/PersonStatus';
import { Business } from './Business';
import { City } from './City';
import { Contacts } from './Contacts';
import { Location } from './Location';
import { Messenger } from './Messenger';
/* tslint:disable:variable-name */

export class Person {
  public Type: PersonStatus;
  public Id: string;
  public Firstname: string;
  public Surname: string;
  public Title: string;
  public Role: string;
  public Picture: boolean;
  public Contacts: Contacts;
  public Location: Location;
  public Business: Business;
  constructor(
    type: PersonStatus,
    id: string,
    firstname: string,
    surname: string,
    title: string,
    role: string,
    picture: boolean,
    _Contacts: Contacts,
    _Location: Location,
    _Business: Business
  ) {
    this.Type = type;
    this.Id = id;
    this.Firstname = firstname;
    this.Surname = surname;
    this.Title = title;
    this.Role = role;
    this.Picture = picture;
    this.Contacts = _Contacts;
    this.Location = _Location;
    this.Business = _Business;
  }

  public isLearner(): boolean {
    return (
      this.Type === PersonStatus.Interner_Lernender || this.Type === PersonStatus.Externer_Lernender
    );
  }
  public isSupervisor(): boolean {
    return this.Role.indexOf('Leiter') >= 0;
  }
  public isAssistent() {
    return this.Role.indexOf('Management & Team Support') >= 0;
  }

  public isOfStatus(status: PersonStatus) {
    return this.Type === status;
  }

  public static empty(): Person {
    return new Person(
      PersonStatus.Interner_Mitarbeiter,
      '',
      '',
      '',
      '',
      '',
      false,
      new Contacts('', '', '', '', new Messenger('', 0)),
      new Location(new City('', ''), [new Room('', '', 0, '', '', '', '', '', '')]),
      new Business([], [], [], [], [], [], '')
    );
  }
}

File: Phonebook.Source.PeopleSoft/src/Phonebook.Source.PeopleSoft.Models/Old/Person.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Phonebook.Source.PeopleSoft.Models.Old {
  public class Person {
    private readonly Models.Person person;

    public Person(Models.Person person) {
      this.person = person;
      this.BuildOldPerson();
    }

    private void BuildOldPerson() {
      Type = new PersonStatus(person.Status).ToString();
      Id = person.ShortName;
      Firstname = person.FirstName;
      Surname = person.LastName;
      Title = person.Title;
      Role = person.Function != null ? person.Function.Label : string.Empty;
      Picture = false;
      Contacts = new Contacts(person);
      if (person.OrgUnit.HeadOfOrgUnitId == person.Id) {
        Role = $"Leiter {person.OrgUnit.Name}";
        Business = new Business(person.OrgUnit, true);
      } else {
        Business = new Business(person.OrgUnit);
      }

      Location = new Location(person);
    }

    public string Type {
      get;
      private set;
    }
    public string Id {
      get;
      private set;
    }
    public string Firstname {
      get;
      private set;
    }
    public string Surname {
      get;
      private set;
    }
    public string Title {
      get;
      private set;
    }
    public string Role {
      get;
      private set;
    }
    public bool Picture {
      get;
      private set;
    }
    public Contacts Contacts {
      get;
      private set;
    }
    public Location Location {
      get;
      private set;
    }
    public Business Business {
      get;
      private set;
    }
  }
}
